resources/indexer.py

```diff
@@ -68,28 +68,28 @@ class ProcessingLevel:
     """Wrapper static class encapsulating processing-level parsing logic and levels constants."""
     PFRaw = "PFRaw"
     PFFilt = "PFFilt"
-    PFDsT = "PFDsT"
+    PFDST = "PFDST"
     L2 = "L2"
     L3 = "L3"
     L4 = "L4"
 
     @staticmethod
-    def from_path(dir_path):
-        """Return the processing level parsed from the dir path, case insensitive."""
-        # Ex: /data/exp/IceCube/2013/filtered/level2/1229/Run00123579_0/
+    def from_filename(path):
+        """Return the processing level parsed from the file name, case insensitive."""
+        # Ex: Level2_IC86.2017_data_Run00130567_Subrun00000000_00000280.i3.zst
         search_strings = {
             "PFRaw": ProcessingLevel.PFRaw,
             "PFFilt": ProcessingLevel.PFFilt,
-            "PFDsT": ProcessingLevel.PFDsT,
+            "PFDST": ProcessingLevel.PFDST,
             "Level2": ProcessingLevel.L2,
             "Level3": ProcessingLevel.L3,
             "Level4": ProcessingLevel.L4
         }
-        dir_path_upper = dir_path.upper()
+        path_upper = path.upper()
         for string, level in search_strings.items():
-            if string.upper() in dir_path_upper:
+            if string.upper() in path_upper:
                 return level
```

Comment:
Can I use this for pDAQ-2ndBld files too?
And can I force it to add a particular file (e.g. a one-off tar file created from many tiny files)?

Reply:
From my reading of the script, these would be handled by the BasicFileMetadata.
This is very similar to the way `ltacmd catalog load` provides the minimum necessary metadata to register the file.
I think as long as the one-off tar was in the directory, it would get picked up.